src/core/client/stream/tabs/Profile/ChangeUsername/UpdateUsernameMutation.tsx

```diff
@@ -0,0 +1,62 @@
+import { graphql } from "react-relay";
+import { Environment } from "relay-runtime";
+
+import {
+  commitMutationPromiseNormalized,
+  createMutation,
+  MutationInput,
+} from "coral-framework/lib/relay";
+import { UpdateUsernameMutation as MutationTypes } from "coral-stream/__generated__/UpdateUsernameMutation.graphql";
+
+let clientMutationId = 0;
+
+const UpdateUsernameMutation = createMutation(
+  "updateUsername",
+  (environment: Environment, input: MutationInput<MutationTypes>) =>
+    commitMutationPromiseNormalized<MutationTypes>(environment, {
+      mutation: graphql`
+        mutation UpdateUsernameMutation($input: UpdateUsernameInput!) {
+          updateUsername(input: $input) {
+            clientMutationId
+            user {
+              username
+              status {
+                username {
+                  history {
+                    username
+                    createdAt
+                  }
+                }
+              }
+            }
+          }
+        }
+      `,
+      variables: {
+        input: {
+          ...input,
+          clientMutationId: (clientMutationId++).toString(),
+        },
+      },
+      optimisticResponse: {
+        updateUsername: {
+          clientMutationId: (clientMutationId++).toString(),
+          user: {
+            username: input.username,
+            status: {
+              username: {
+                history: [
```

Comment:
A user could have previously had a username history, you should merge their existing username history here.

Reply:
does that mean I should add the username history to the `UpdateUsernameInput` type in the schema? or create a different interface to use here